cmd/skaffold/app/cmd/flags.go

```diff
@@ -432,10 +432,15 @@ var flagRegistry = []Flag{
 		DefinedOn:     []string{"dev", "debug"},
 	},
 	{
-		Name:          "add-skaffold-labels",
-		Usage:         "Add Skaffold-specific labels to rendered manifest. Custom labels will still be applied. Helpful for GitOps model where Skaffold is not the deployer.",
-		Value:         &opts.AddSkaffoldLabels,
-		DefValue:      false,
+		Name:     "add-skaffold-labels",
+		Usage:    "Add Skaffold-specific labels to rendered manifest. Custom labels will still be applied. Helpful for GitOps model where Skaffold is not the deployer.",
+		Value:    &opts.AddSkaffoldLabels,
+		DefValue: false,
+		DefValuePerCommand: map[string]interface{}{
+			"dev":   true,
+			"debug": true,
+			"run":   true,
+		},
 		FlagAddMethod: "BoolVar",
 		DefinedOn:     []string{"render"},
```

Comment:
You need to add `dev`, `debug`, and `run` here.  It might make sense to invert the defaults so `DevValuePerCommand = { "render": false }`.

Reply:
> You need to add dev, debug, and run here. 

I just updated this, but after pushing the change I'm thinking we don't actually want to define this for `dev`/`debug`/`run` here - do we really want this flag to be available for use on those commands? `skaffold` won't work correctly if the user messes with it.

> It might make sense to invert the defaults so DevValuePerCommand = { "render": false }.

regardless of my above comment, agree with this

> You should need to regenerate the man pages when changing the defaults.

I think we also don't need to do this, since this flag is `Deprecated` it won't show up in the help text.